### DISCUSSION TITLE
Trouble installing anaconda on new M1 mac

### DISCUSSION THREAD

**Discussion by masonrogers14** [Homebrew/discussions#3308] [Casks]
I just tried installing anaconda on my new M1 mac with `brew install --cask anaconda`. The install failed with
```
/opt/homebrew/Caskroom/anaconda/2022.05/Anaconda3-2022.05-MacOSX-arm64.sh: line 405: /opt/homebrew/anaconda3/conda.exe: Bad CPU type in executable
/opt/homebrew/Caskroom/anaconda/2022.05/Anaconda3-2022.05-MacOSX-arm64.sh: line 407: /opt/homebrew/anaconda3/conda.exe: Bad CPU type in executable
```
I see that native support for M1 is [new to the 05.2022 version of anaconda](https://www.anaconda.com/blog/new-release-anaconda-distribution-now-supporting-m1), so I am surprised to get a 'bad CPU type' error. I tried the standard fixes--homebrew is up to date, `brew doctor` reports nothing wrong, etc. I am curious if anyone has troubleshooting suggestions or if there may be a bug with the new anaconda version.

Thank you!

**Comment by gromgit** [comment]
`brew install --cask anaconda` works perfectly on my M1 Mac mini, so it could be a transient problem. Try reinstalling as follows:
```sh
brew uninstall --cask anaconda   # It's OK if this errors out
rm -fr "$(brew --cache)"   # Wipe Homebrew cache, to avoid reinstalling a b0rked download
brew install --cask anaconda
```

**Reply by masonrogers14** [reply]
Thank you for the advice; unfortunately the problem persists for me. May I ask which version of macOS you have? I am running 12.2.1.

**Reply by gromgit** [reply]
I'm running 12.3.1, but it's unlikely to be an issue.

What's the output of `/usr/bin/shasum -a 256 /opt/homebrew/Caskroom/anaconda/2022.05/Anaconda3-2022.05-MacOSX-arm64.sh`? If it's anything other than `457b10272f807879ab4289ffb93d7dcb695362b417bb7c80d24892c8d8557b29`, your download is getting b0rked after Homebrew's done its checksumming.

**Reply by masonrogers14** [reply]
Checksum is correct. I'm also having the same issue trying with the anaconda installer

**Reply by gromgit** [reply]
Just to be sure, what's the output of `brew config` and `brew doctor`?

**Reply by masonrogers14** [reply]
```
brew config
HOMEBREW_VERSION: 3.4.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 30e3d8883cde265cbc16ff94b49a3ac14137590e
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: cc70d0d9cb9b2405837290118c8a8d55cb65d563
Core tap last commit: 38 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2.1-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: false
```

and

```
brew doctor
Your system is ready to brew.
```

**Reply by gromgit** [reply]
Nothing seems out of the ordinary, and if you're getting the same problem with the official installer, it's the installer that's interacting badly with your setup (the cask basically downloads and runs the installer). You should probably ask the Anaconda folks for help.

**Reply by masonrogers14** [reply]
Okay, thanks so much for your help :)

**Comment by section83** [comment]
The "Bad CPU type in executable" error normally occurs when attempting to execute Intel code and Rosetta 2 is not installed.  Try running an Intel-only app and then agree to Rosetta 2 being installed.  I use VLC-intel for that purpose.

![macos-big-sur-software-update-rosetta-alert](https://user-images.githubusercontent.com/7471595/170806110-68205705-cfac-428e-b5d6-0b0a519fef56.jpg)

See: https://support.apple.com/en-gb/HT211861.

**Reply by dukechain2333** [reply]
Try install Rosetta first.

**Reply by masonrogers14** [reply]
Thanks for the advice--I'd prefer not to install Rosetta. Anaconda describes this release as featuring native M1 support, but I guess the support either isn't 100% or there's something weird about my setup. I've proceeded with miniforge for the time being.

**Reply by dukechain2333** [reply]
I just bought my M1 Pro mbp yesterday and also meet this problem. I guess the most weird thing is that if you download the exact same version from Anaconda official website, the install program will suggest you to install Rosetta first.

**Comment by gah-bo** [comment]
To be clear, my M1 is my first Apple product and I got it a month ago. I don't know about homebrew and have never used it. 

That being said, I had a similar issue using the GUI installer - Anaconda would install as Kind Intel. 

Solution: Use the terminal installer, not GUI.  Another difference is I installed in Users/myname/anaconda3/ rather than /opt/anaconda3/ which I was doing with GUI. Unknown if this makes a difference. I just tried to test it but I don't have perms to create the anaconda3 folder within opt, so I will leave it here.

Unknown whether kind being Apple Silicon means I can't use packages that lack an ARM version. I am also confused about the fact that blogpost for 2022-05 update states M1 won't support Anaconda Navigator yet - mine works perfectly (both when I installed through GUI as Intel, and now through terminal as Apple Silicon)

(this is my answer for https://datascience.stackexchange.com/questions/111507/new-anaconda-release-for-m1-2022-05 which is a similar issue)

![image](https://user-images.githubusercontent.com/64602073/172078740-ffb39521-2a23-4af8-9129-0918cdb68f7a.png)
